backend/app/adapter/routing/handle/search.go

```diff
@@ -87,24 +93,28 @@ func Search(
 		}
 		filter, err := search.NewFilter(body.Filter.MaxResults, body.Filter.Resources, body.Filter.Orders)
 		if err != nil {
+			i.SearchFailed(err)
 			http.Error(w, err.Error(), http.StatusInternalServerError)
 			return
 		}
 
 		results, err := searcher.Search(query, filter)
 		if err != nil {
+			i.SearchFailed(err)
 			http.Error(w, err.Error(), http.StatusInternalServerError)
 			return
 		}
 
 		response := newSearchResponse(results)
 		respBody, err := json.Marshal(&response)
 		if err != nil {
+			i.SearchFailed(err)
 			http.Error(w, err.Error(), http.StatusInternalServerError)
 			return
 		}
 
 		w.Write(respBody)
+		i.SearchSucceed()
```

Comment:
We need to track the keyword being searched, in which city and at what time so that we can analyze and determine the new features we want to build in order to better serve the users. This means we need to include analytics.